src/cgroups/v1/memory.rs

```diff
@@ -214,11 +214,11 @@ impl Memory {
                 let current_limit = Self::get_memory_limit(cgroup_root)?;
                 match resource.swap {
                     Some(swap) => {
-                        let is_updated = swap == -1 || current_limit < swap;
+                        let is_updated = swap <= -1 || current_limit < swap;
```

Comment:
I think this change is a good thing. But have there actually been any such a case?

Reply:
I think it matches the test case better, but either way the tests pass. I think it might actually be useful here to see what the subsystem thinks of negative values. I'm not sure if the kernel will give a validation error for negative values less than `-1` or if all negative values are considered `unlimited`. If it's the case that all negative values are `unlimited` then I would say this changed behavior is most correct, otherwise I'd say any negative value less than `-1` should cause it's own expected failure and the tests should account for that. I'll look into this deeper, and leave a comment in the code to explain.

Reply:
I decided to revert this. So if you set any of these below -1 it will result in an "Invalid Arguments" error, which will be the same thing that happens in Rust. We don't currently deal with any of the validation error's the kernel spits out, so I decided to just leave it. Eventually it might be nice to handle the error in the controller code and create our own set of errors to more reasonably represent what the error actually means, and provide some feedback of what the valid values actually are.